src/python/foglamp/data_purge/sqlalchemy_purge.py

```diff
@@ -0,0 +1,324 @@
+    """
+    rand = random.randint(1, 100)
+
+    stmt = "SELECT id FROM (SELECT id FROM readings ORDER BY id ASC LIMIT %s)t ORDER BY id DESC LIMIT 1"
+    row_id = get_count(stmt % rand)
+
+    with open(config_file, 'r') as conf:
+        config_info = json.load(conf)
+
+    config_info["lastID"] = row_id
+    open(config_file, 'w').close()
+    with open(config_file, 'r+') as conf:
+        conf.write(json.dumps(config_info))
+
+"""The actual purge process 
+"""
+
+
+def purge_process_function(table_name) -> int:
+    """The actual process read the configuration file, and based off the information in it does the following:
+    1. Gets previous information found in log file
+    2. Based on the configurations, call the DELETE command to purge the data
+    3. Calculate relevant information kept in logs
+    4. Based on the configuration calculates how long to wait until next purge, and returns that
+    
+    Args: 
+        table_name (SQLAlchemy.Table): The name of the table queries run against
+    Returns: 
+        Amount of time until next purge process
+    """
+
+    # Reload config (JSON File) - age,  enabled,  wait,  pi_date
+    with open(config_file, 'r') as conf:
+        config = json.load(conf)
+
+    data = {}
+    purge_status = {}
+
+    if os.path.getsize(logs_file) > 0:
+        with open(logs_file,  'r') as f:
+            data = json.load(f)
+
+    if config['enabled'] is True:  # meaning that config info is authorizing the purge
+        purge_status['startTime'] = datetime.datetime.fromtimestamp(time.time())
+        age_timestamp = datetime.datetime.strftime(purge_status['startTime'] - convert_timestamp(
+            set_time=config['age']), '%Y-%m-%d %H:%M:%S.%f')
+        purge_status['startTime'] = purge_status['startTime'].strftime('%Y-%m-%d %H:%M:%S')
+        last_connection_id = config['lastID']
+
+        # Number of rows exist at the point of calling purge
+        total_count_before = get_count(sqlalchemy.select([sqlalchemy.func.count()]).select_from(table_name).where(
+            table_name.c.ts < purge_status['startTime']))
+
+        # Number of unsent rows
+        number_sent_rows = sqlalchemy.select([sqlalchemy.func.count()]).select_from(table_name).where(
+            table_name.c.id <= last_connection_id).where(table_name.c.ts < purge_status['startTime'])
+        unsent_rows_before = total_count_before-get_count(number_sent_rows)
+
+        """Time purge process starts
+        If unsent data is retained, then the WHERE condition is against the last sent ID
+        """
+        unsent_rows_removed_query = sqlalchemy.select([sqlalchemy.func.count()]).select_from(table_name).where(
+            table_name.c.id > last_connection_id).where(table_name.c.ts <= age_timestamp)
+
+        unsent_rows_removed_before = get_count(unsent_rows_removed_query)
+        if config['retainUnsent'] is True:
+            delete_query = sqlalchemy.delete(table_name).where(table_name.c.id <= last_connection_id).where(
```

Comment:
OK, it looks like the code does the right thing, it was just the comment above that was misleading.